sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SemanticModelHelper.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class SemanticModelHelper
+    {
+        public static SemanticModel GetCorrectSemanticModel(this SemanticModel model, SyntaxNode node)
```

Comment:
May be this should be `ISymbol GetDeclaredSymbol(this SemanticModel, SyntaxNode)` and `ISymbol GetSymbol(this SemanticModel, SyntaxNode)`, because usually you obtain the semantic model to get the symbol and why not always get the correct symbol? The problem is GetDeclaredSymbol has many overloads and also we should think of names that do not collide with the existing instance methods.
In the worst case I would rename the method to `GetSemanticModel`, "correct" does not sound right here, as if Roslyn is lying to us (which it does)...